client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	getShortCodeAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+		const defaultWidth = 640;
+		const defaultHeight = defaultWidth * 9 / 16;
+		return {
```

Comment:
>Very handy, indeed! It makes the intent of the code much clearer.

Actually, this will not work as expected, as the presence of the key in the second argument will override the default value, regardless of whether the value is truthy.

For example, try entering this in your browser console (assuming your browser supports `Object.assign`):

```js
Object.assign( { a: 1 }, { a: undefined } );
```